lib/vsc/utils/generaloption.py

```diff
@@ -222,7 +223,13 @@ def take_action(self, action, dest, opt, value, values, parser):
         """Extended take_action"""
         orig_action = action  # keep copy
 
-        if action == 'shorthelp':
+        # dest is None for actions like shorthelp and confighelp
+        if dest and getattr(parser._long_opt.get('--' + dest, ''), 'store_or', '') == 'help':
+            Option.take_action(self, action, dest, opt, value, values, parser)
+            fn = getattr(parser, 'print_%shelp' % values.help)
+            fn()
+            parser.exit()
+        elif action == 'shorthelp':
```

Comment:
open an issue to convert `shorthelp` and `confighelp` in aliases of resp `help=short` and `help=config` (it's not straightforward, so a bit out of the scope of this PR)